### PR TITLE
Refactored XBlockAside rendering and added support for student view

### PR DESCRIPTION
**Pull request by gsidebo**
This PR adds support for aside rendering in the LMS student view, and refactors the aside code to use the same code path in LMS/Studio for dispatching requests to the handlers defined in the aside.

### CODE REVIEW

**Comment by gsidebo** [comment]
@noisecapella This is ready for review now. I figured out the JS errors and the repeated click handler. The barebones rapid response aside in my other PR (https://github.com/mitodl/rapid-response-xblock/pull/18) should be sufficient to test.

File: common/static/common/js/xblock/core.js [hunk 1]
```diff
@@ -11,6 +11,9 @@
         } else {
             selector = '.' + blockClass;
         }
+        // After an element is initialized, a class is added to it. To avoid repeat initialization, no
+        // elements with that class should be selected.
+        selector += ':not(.xblock-initialized)';
```
Comment (noisecapella) [inline common/static/common/js/xblock/core.js:16]: Does this fix that JS bug you found?

File: lms/djangoapps/courseware/module_render.py [hunk 2]
```diff
@@ -61,7 +62,9 @@
     replace_course_urls,
     replace_jump_to_id_urls,
     replace_static_urls,
-    wrap_xblock
+    wrap_xblock,
+    wrap_xblock_aside,
```
Comment (noisecapella) [inline lms/djangoapps/courseware/module_render.py]: I think this is an unused import

File: openedx/core/lib/xblock_utils/__init__.py [hunk 3]
```diff
@@ -500,3 +501,32 @@ def xblock_resource_pkg(block):
...
+    """
+    return isinstance(usage_key, (AsideUsageKeyV1, AsideUsageKeyV2))
+
+
+def get_aside_from_xblock(xblock_descriptor, aside_type):
+    """
+    Gets an instance of an XBlock aside from the XBlock that it's decorating
+
+    Args:
+        xblock_descriptor: The descriptor of the XBlock that the desired aside is decorating
```
Comment (noisecapella) [inline openedx/core/lib/xblock_utils/__init__.py]: Can you add a type here?

File: openedx/core/lib/xblock_utils/__init__.py [hunk 4]
```diff
@@ -500,3 +501,32 @@ def xblock_resource_pkg(block):
...
+        aside_type (str): The aside type
+
+    Returns:
+        Instance of an xblock aside
```
Comment (noisecapella) [inline openedx/core/lib/xblock_utils/__init__.py]: Can you add a type here?

**Review by noisecapella — CHANGES_REQUESTED** [review]
Looks good overall just some minor comments. Also I think you'll need to add a test to verify the CMS view works as expected for asides

File: common/static/common/js/xblock/core.js (hunk 1 above)
Comment (gsidebo) [reply common/static/common/js/xblock/core.js:16]: Yup. The aside renders a certain element, and that element was being selected via `.immediateDescendents(selector)` for two different elements, and thus the click handler was being attached twice. This selector change makes sure that any element that has already been initialized will not be selected again

File: openedx/core/lib/xblock_utils/__init__.py [hunk 5]
```diff
@@ -500,3 +501,32 @@ def xblock_resource_pkg(block):
...
+    """
+    return isinstance(usage_key, (AsideUsageKeyV1, AsideUsageKeyV2))
+
+
+def get_aside_from_xblock(xblock, aside_type):
+    """
+    Gets an instance of an XBlock aside from the XBlock that it's decorating
+
+    Args:
+        xblock (xblock.core.XBlock): The XBlock that the desired aside is decorating
```
Comment (gsidebo) [inline openedx/core/lib/xblock_utils/__init__.py:524]: Made enough sense to expect an `XBlock` type here. That class inherits from `RuntimeServicesMixin`, and 
this function's only client right now is passing in an `XModuleDescriptor`, which inherits from `XBlock`

**Review by noisecapella — APPROVED** [review]

**Comment by pdpinch** [comment]
Should this be opened upstream? We won't be releasing this aside into Residential MITx production until the Hawthorn release.

**Comment by gsidebo** [comment]
@pdpinch We can put up a PR, but there are a couple concerns: (1) We still haven't seen the only known aside (`StructuredTagsAside`) in action, so it would be on them to make sure that still works. Adds a little to the testing burden; (2) This PR fixes the student view for asides, but there isn't an actual aside implementation that makes use of the student view (until ours is live, of course). It's possible that they wouldn't want to take the time to review for a feature that isn't relevant to them yet. I'm happy to just open it up and see what happens though

**Pull request merged**